src/components/flame-graph/MaybeFlameGraph.js

```diff
@@ -52,7 +68,7 @@ class MaybeFlameGraph extends React.PureComponent<Props> {
         </div>
       );
     }
-    return <FlameGraph />;
+    return <FlameGraph viewportRef={this._takeViewportRef} />;
```

Comment:
instead of passing the viewport's ref like this which seems a bit fragile, what do you think of implementing a `focus` function on the FlameGraph component that we could call here?

Reply:
That would be much clearer! I'll take a look :+1: